### PR TITLE
feat: using cloud-init in StackScript

### LINKED ISSUE #7: install cloud-init using stackscript and run the yaml cloud init file.
Linode image does not have cloud-init. Therefore we should install it with stackscript to use the same cloud-init yaml config file.

Please write the stackscript to do that, and use it with your python code.
Also write appropriate documentation.

### PR DESCRIPTION
**Pull request by dr-mato**

### CODE REVIEW

**Comment by dr-mato** [comment]
#7

File: app/stackscript.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+apt update
+apt upgrade -y
+apt install default-jre  git -y
+git clone https://github.com/dr-mato/minecraft_server.git
+FILE_USERDATA="/var/lib/cloud/seed/nocloud-net/user-data"
```
Comment (samerbahri98) [inline app/stackscript.txt]: change line 6 to
```shell
FILE_USERDATA=/etc/cloud/cloud.cfg.d/minecraft.cfg
```

File: app/cloud-config.yml
Comment (samerbahri98) [inline app/cloud-config.yml:1]: We need to wait for @ntatrishvili  to fix her cloudinit file and use it instead.
I will make the appropriate suggestions when she does that. until then use her file as it is
[>here](https://github.com/ntatrishvili/minecraft_server/blob/17cd9918c72d5bf3541a6fe09bf30f7ea33bed45/iac/cloudinit/cloud-init.yml)

File: app/stackscript.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+apt update
+apt upgrade -y
+apt install default-jre  git -y
```
Comment (samerbahri98) [inline app/stackscript.txt:5]: you can install cloud-init here too, instead of the end of the file.

File: app/stackscript.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+warnings:
+  dsid_missing_source: off
+EOS
+UMASK=$(umask)
+umask 0066
+echo "Creating $FILE_USERDATA..."
+touch "${FILE_USERDATA}"
+cat ./minecraft_server/app/cloud-config.yml > "${FILE_USERDATA}"
+umask "${UMASK}"
+apt-get -q=2 update
```
Comment (samerbahri98) [inline app/stackscript.txt]: you don't need this if you install cloud init at the beginning

File: app/stackscript.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (samerbahri98) [inline app/stackscript.txt]: #! /bin/bash

File: app/stackscript.txt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+EOS
+UMASK=$(umask)
+umask 0066
+echo "Creating $FILE_USERDATA..."
+touch "${FILE_USERDATA}"
+cat ./minecraft_server/app/cloud-config.yml > "${FILE_USERDATA}"
+umask "${UMASK}"
+apt-get -q=2 update
+apt-get -q=2 install -y cloud-init
+cloud-init init
```
Comment (samerbahri98) [inline app/stackscript.txt:18]:
```
cloud-init init --local
```

File: app/stackscript.txt
Comment (samerbahri98) [inline app/stackscript.txt:1]: if the stackscript fails, you can ssh into the server, copy this file there and run it as a shell script.

File: iac/python/createLinodeServer.py
Comment (samerbahri98) [inline iac/python/createLinodeServer.py:1]: use an environment variable for the root_pass. never commit passwords or sensitive information to a git repository, especially a public one.

File: iac/python/createMinecraftServer.py
Comment (samerbahri98) [inline iac/python/createMinecraftServer.py:1]: same comment as `iac/python/createLinodeServer.py`

**Review by samerbahri98 — CHANGES_REQUESTED** [review]
for now just copy and paste the cloud-init file from @ntatrishvili's branch. once that is merged I will suggest further improvements.

**Comment by samerbahri98** [comment]
@dr-mato @ntatrishvili any update on this?

File: app/stackscript.txt (hunk 2 above)
Comment (samerbahri98) [reply app/stackscript.txt:5]: This still missing

File: app/stackscript.txt (hunk 5 above)
Comment (samerbahri98) [reply app/stackscript.txt:18]: the local flag is missing

File: app/stackscript.txt [hunk 6]
```diff
@@ -1,24 +1,20 @@
...
 #cloud-config
 warnings:
   dsid_missing_source: off
 EOS
-UMASK=$(umask)
-umask 0066
 echo "Creating $FILE_USERDATA..."
-touch "${FILE_USERDATA}"
-cat ./minecraft_server/app/cloud-config.yml > "${FILE_USERDATA}"
-umask "${UMASK}"
```
Comment (samerbahri98) [inline app/stackscript.txt]: we will wait for @ntatrishvili  for this section

**Review by samerbahri98 — CHANGES_REQUESTED** [review]

**Comment by samerbahri98** [comment]
Cancelled Because of Digitalocean Migration.

**Pull request closed without merging**